processing/dwf_calculation_algorithm.py

```diff
@@ -0,0 +1,396 @@
+    algorithms and explain methods and variables used to do it. An
+    algorithm like this will be available in all elements, and there
+    is not need for additional work.
+
+    All Processing algorithms should extend the QgsProcessingAlgorithm
+    class.
+    """
+
+    # Constants used to refer to parameters and outputs. They will be
+    # used when calling the algorithm from another algorithm, or when
+    # calling from the QGIS console.
+
+    OUTPUT = 'OUTPUT'
+    INPUT = 'INPUT'
+
+    def initAlgorithm(self, config):
+        """
+        Here we define the inputs and output of the algorithm, along
+        with some other properties.
+        """
+
+        # We add the input vector features source. It can have any kind of
+        # geometry.
+        self.addParameter(
+            QgsProcessingParameterProviderConnection(
+                name = self.INPUT,
+                description = self.tr('Input spatialite (.sqlite)'),
+                provider = 'spatialite'
+            )
+        )
+        
+        self.addParameter(
+            QgsProcessingParameterString(
+                'start_time',
+                self.tr('Start time of day (HH:MM:SS)'),
+                '00:00:00'
+            )
+        )
+                    
+        self.addParameter(
+            QgsProcessingParameterString(
+                'duration',
+                self.tr('Simulation duration (seconds)')
+            )
+        )
+        
+        
+        self.addParameter(
+            QgsProcessingParameterFile(
+                'dwf_progress_file',
+                self.tr('DWF progress file (.csv)'),
+                extension = 'csv',
+                defaultValue = None,
+                optional = True
+            )
+        )
+
+
+        self.addParameter(
+            QgsProcessingParameterFileDestination(
+                self.OUTPUT,
+                self.tr('Output CSV'),
+                'csv(*.csv)'
+            )
+        )
+        
+        
+
+    def processAlgorithm(self, parameters, context, feedback):
+        """
+        Here is where the processing itself takes place.
+        """
+        
+        start_time = self.parameterAsString(parameters, 'start_time', context)
+        duration = self.parameterAsDouble(parameters, 'duration', context)
+        connection_name = self.parameterAsConnectionName(parameters, self.INPUT, context)
+        output_csv = self.parameterAsFileOutput(parameters, self.OUTPUT, context)
+        dwf_factor_input = self.parameterAsFile(parameters, 'dwf_progress_file', context)
+
+        try:
+            md = QgsProviderRegistry.instance().providerMetadata('spatialite')
+            conn = md.createConnection(connection_name)
+        except QgsProviderConnectionException:
+            raise QgsProcessingException(self.tr('Could not retrieve connection details for {}').format(connection_name))
+            
+        spatialite_fn = conn.uri()[8:-1]
```

Comment:
I first thought "spatialite function", but later it seemed to be "spatialite filename" => no abbreviations.